Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm

```diff
@@ -514,65 +521,90 @@
 
 double PlaybackSessionModelMediaElement::duration() const
 {
-    return m_mediaElement ? m_mediaElement->duration() : 0;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->duration();
+    return 0;
 }
 
 double PlaybackSessionModelMediaElement::currentTime() const
 {
-    return m_mediaElement ? m_mediaElement->currentTime() : 0;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->currentTime();
+    return 0;
 }
 
 double PlaybackSessionModelMediaElement::bufferedTime() const
 {
-    return m_mediaElement ? m_mediaElement->maxBufferedTime() : 0;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->maxBufferedTime();
+    return 0;
 }
 
 bool PlaybackSessionModelMediaElement::isPlaying() const
 {
-    return m_mediaElement ? !m_mediaElement->paused() : false;
+    if (RefPtr mediaElement = m_mediaElement)
+        return !mediaElement->paused();
+    return false;
 }
 
 bool PlaybackSessionModelMediaElement::isStalled() const
 {
-    return m_mediaElement && m_mediaElement->readyState() <= HTMLMediaElement::HAVE_CURRENT_DATA;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->readyState() <= HTMLMediaElement::HAVE_CURRENT_DATA;
+    return false;
 }
 
 double PlaybackSessionModelMediaElement::defaultPlaybackRate() const
 {
-    return m_mediaElement ? m_mediaElement->defaultPlaybackRate() : 0;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->defaultPlaybackRate();
+    return 0;
 }
 
 double PlaybackSessionModelMediaElement::playbackRate() const
 {
-    return m_mediaElement ? m_mediaElement->playbackRate() : 0;
+    if (RefPtr mediaElement = m_mediaElement)
+        return mediaElement->playbackRate();
+    return 0;
 }
 
 Ref<TimeRanges> PlaybackSessionModelMediaElement::seekableRanges() const
 {
-    return m_mediaElement && m_mediaElement->supportsSeeking() ? m_mediaElement->seekable() : TimeRanges::create();
+    if (RefPtr mediaElement = m_mediaElement) {
```

Comment:
this could be done more concisely as:
```
if (RefPtr mediaElement = m_mediaElement; mediaElement && mediaElement->supportsSeeking())
    return mediaElement->seekable();
```